test/client.test.js

```diff
@@ -13,18 +13,42 @@ describe('Client', () => {
       expect(client.liveServiceHostname).toEqual('https://kls.kinvey.com');
     });
 
-    it('should be able to provide custom apiHostname', () => {
+    it('should be able to provide custom apiHostname with protocol https:', () => {
       const apiHostname = 'https://mybaas.kinvey.com';
       const client = new Client({ apiHostname: apiHostname });
       expect(client.apiHostname).toEqual(apiHostname);
     });
 
-    it('should be able to provide custom micHostname', () => {
+    it('should be able to provide custom apiHostname with protocol http:', () => {
+      const apiHostname = 'http://mybaas.kinvey.com';
+      const client = new Client({ apiHostname: apiHostname });
+      expect(client.apiHostname).toEqual(apiHostname);
```

Comment:
@thomasconner this test is potentially missing that the `apiProtocol` in this case will be `https://` because it was added by the SDK.